.github/workflows/issues.yml

```diff
@@ -12,122 +12,34 @@ jobs:
-            github.rest.issues.addAssignees({
-              issue_number: context.issue.number,
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              assignees: ['bschnurr']
-            })
-      - name: Sunday (Erik)
-        if: steps.proceed.outputs.result == 1 && steps.day.outputs.number == 0
-        uses: actions/github-script@v6.3.3
-        with:
-          github-token: ${{secrets.GITHUB_TOKEN}}
-          script: |
-            github.rest.issues.addAssignees({
-              issue_number: context.issue.number,
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              assignees: ['debonte']
-            })
-      - name: Monday (Heejae)
-        if: steps.proceed.outputs.result == 1 && steps.day.outputs.number == 1 && steps.hour.outputs.hour < 16
-        uses: actions/github-script@v6.3.3
-        with:
-          github-token: ${{secrets.GITHUB_TOKEN}}
-          script: |
-            github.rest.issues.addAssignees({
-              issue_number: context.issue.number,
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              assignees: ['heejaechang']
-            })
-      - name: Tuesday (Jude)
-        if: steps.proceed.outputs.result == 1 && (steps.day.outputs.number == 1 && steps.hour.outputs.hour >= 16) || (steps.day.outputs.number == 2 && steps.hour.outputs.hour < 16)
+      - name: Find last assigned
+        id: assigned
         uses: actions/github-script@v6.3.3
         with:
           github-token: ${{secrets.GITHUB_TOKEN}}
+          result-encoding: string
           script: |
-            github.rest.issues.addAssignees({
-              issue_number: context.issue.number,
+            const issues = await github.rest.issues.listForRepo({
+              state: 'all',
+              assignee: '*',
               owner: context.repo.owner,
-              repo: context.repo.repo,
-              assignees: ['judej']
-            })
-      - name: Wednesday (Rich)
-        if: steps.proceed.outputs.result == 1 && (steps.day.outputs.number == 2 && steps.hour.outputs.hour >= 16) || (steps.day.outputs.number == 3 && steps.hour.outputs.hour < 16)
-        uses: actions/github-script@v6.3.3
-        with:
-          github-token: ${{secrets.GITHUB_TOKEN}}
-          script: |
-            github.rest.issues.addAssignees({
-              issue_number: context.issue.number,
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              assignees: ['rchiodo']
-            })
-      - name: Thursday (Rich)
-        if: steps.proceed.outputs.result == 1 && (steps.day.outputs.number == 3 && steps.hour.outputs.hour >= 16) || (steps.day.outputs.number == 4 && steps.hour.outputs.hour < 16)
-        uses: actions/github-script@v6.3.3
+              repo: context.repo.repo
+            });
+            return issues.data.length > 0 ? issues.data[0].assignee.login : '';
```

Comment:
I figured out how to run this locally, so it was easier to test. You can do this locally by npm installing @octokit/rest and then making a js file that does this same thing, it just needs a auth token to work.

Reply:
demo this please

Reply:
Here's the source that I used to test this:

```javascript
const { Octokit } = require('@octokit/rest');

async function run() {
    const octokit = new Octokit({
        auth: '<personal access token>',
        userInfo: 'rchiodo',
    });
    const issues = await octokit.rest.issues.listForRepo({
        state: 'all',
        assignee: '*',
        repo: 'pyrx',
        owner: 'microsoft',
    });
    console.log(JSON.stringify(issues.data[0], undefined, 2));
}

run();

```